stl.go

```diff
@@ -774,12 +803,23 @@ func (s Subtitles) WriteToSTL(o io.Writer) (err error) {
 		return
 	}
 
-	// Loop through items
-	for idx, item := range s.Items {
-		// Write tti block
-		if _, err = o.Write(newTTIBlock(item, idx+1).bytes(g)); err != nil {
-			err = fmt.Errorf("astisub: writing tti block #%d failed: %w", idx+1, err)
-			return
+	if s.TTIBlocks != nil && len(s.TTIBlocks) == len(s.Items) {
```

Comment:
You don't need this change with `ItemMetadata`.

Reply:
Done